docs/sources/index.md

```diff
@@ -293,5 +292,44 @@ TypesafeConfig.fromHoconString(jsonString, automaticDescription)
 
 
 ```
-
 Please check other ways to load the hocon file in `TypesafeConfig`
+
+## Command Line Arguments Source
+Simply source config from command line arguments
+```scala mdoc:silent
+final case class SimpleCommandLineConfig(key1: String, key2: String)
+
+val simpleCmdLineArgs = "--key1 value1 --key2 value2"
+val simpleSource = ConfigSource.fromCommandLineArgs(simpleCmdLineArgs.split(' ').toList)
+val simpleConfig = descriptor[SimpleCommandLineConfig] from simpleSource
+```
+With nested path in command line arguments
+```scala mdoc:silent
+final case class SparkConf(key1: String, key2: String)
+final case class NestedCommandLineConfig(conf: SparkConf, key3: String)
+
+val nestedCmdLineArgs = "--conf.key1 v1 --conf.key2 v2 --key3 v3"
+val nestedSource = ConfigSource.fromCommandLineArgs(nestedCmdLineArgs.split(' ').toList, keyDelimiter = Some('.'))
+val nestedConfig = descriptor[NestedCommandLineConfig] from nestedSource
+```
+The config with a complex source of command line arguments is what every production service needs
+```scala mdoc:silent
+final case class UserPassword(username: String, password: String)
+final case class DatabaseConfig(database: UserPassword, url: String)
+final case class VaultConfig(userPassword: UserPassword)
+final case class SparkConfig(databaseConfig: DatabaseConfig, numexecs: Int)
+final case class AppConfig(conf: SparkConfig, vault: VaultConfig, users: List[String], region: List[String])
+
+val complexArgs = "--conf.database.username=Foo --conf.database.password=Bar --conf.database.url=jdbc://xyz --conf.numexecs=10 --vault.username=Foo2 --vault.password=Bar2 --users Jane --users Jack --region TW,US"
+
+val complexSource = ConfigSource.fromCommandLineArgs(
+  complexArgs.split(' ').toList,
+  Some('.'),
+  Some(',')
+)
+val appConfig = descriptor[AppConfig] from complexSource
+```
+* From above, command line arguments `--users 1 --users 2` is the source of `users: List[String]` config.
+    * A ConfigDescriptor `list(url)(string)` returns all the urls with command line arguments: `—url bla.com —url xyz.com —url abc.com`  
+* With an assisted delimiter `','`, you can source `region: List[String]` from command line arguments: `--region 111,112`.
+* For nested path, `val args = "—conf -database.username=1"` can be another way to nest, and using `map(conf)(string)` can retrieve all the key-value pairs under `conf: SparkConfig`.
```

Comment:
Since we already explained about list, map etc before, we don't need much explanation here.
Remove line number 332 to 335